core/trino-main/src/main/java/io/trino/sql/gen/IfCodeGenerator.java

```diff
@@ -50,9 +52,10 @@ public BytecodeNode generateExpression(BytecodeGeneratorContext context)
         BytecodeBlock conditionBlock = new BytecodeBlock()
                 .append(context.generate(condition))
                 .comment("... and condition value was not null")
-                .append(wasNull)
-                .invokeStatic(CompilerOperations.class, "not", boolean.class, boolean.class)
-                .invokeStatic(CompilerOperations.class, "and", boolean.class, boolean.class, boolean.class)
+                .append(not(wasNull))
```

Comment:
are there tests that cover code generated by this and other changes?
Are all cases covered (`condition`, `wasNull` = `true` and`false`)?

Reply:
Not sure now, I'll take a look when I have time (maybe this weekend)